packages/wrangler/src/__tests__/jest.setup.ts

```diff
@@ -93,33 +93,36 @@ jest.mock("ws", () => {
 jest.mock("undici", () => {
 	return {
 		...jest.requireActual("undici"),
-		fetch: jest.requireActual("jest-fetch-mock"),
+		/**
+		 * So... Why do we have this hacky mock?
+		 * First, the requirements that necessitated it (if you're looking at this code in horror at some point in the future and these no longer apply, feel free to adjust this implementation!)
+		 * - Wrangler supports Node v16. Once Wrangler only supports Node v18 we can use globalThis.fetch directly and remove this hack
+		 * - MSW makes it difficult to use custom interceptors, and _really_ wants you to use globalThis.fetch. In particular, it doesn't support intercepting undici.fetch
+		 * Because Wrangler supports Node v16, we have to use undici's fetch directly rather than using globalThis.fetch. We'd also like to intercept requests with MSW
+		 * Therefore, we mock undici in tests to replace the imported fetch with globalThis.fetch (which MSW will replace with a mocked version—hence the getter, so that we always get the up to date mocked version)
```

Comment:
Perhaps additionally we can say that we can delegate to global fetch here because we actually run our tests in node 18